examples/graphics/video-texture.html

```diff
@@ -0,0 +1,137 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>PlayCanvas Video Texture</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+    <link rel="icon" type="image/png" href="../playcanvas-favicon.png" />
+    <script src="../lib/playcanvas.js"></script>
+    <script src="../lib/playcanvas-extras.js"></script>
+    <style>
+        body { 
+            margin: 0;
+            overflow-y: hidden;
+        }
+    </style>
+</head>
+
+<body>
+    <!-- The canvas element -->
+    <canvas id="application-canvas"></canvas>
+
+    <!-- Video element -->
+    <video id="vid" autoplay loop muted playsinline crossorigin="anonymous" style="display:none">
+        <source src="../assets/video/SampleVideo_1280x720_1mb.mp4" type="video/mp4">
+    </video>
+
+    <!-- The script -->
+    <script>
+        var canvas = document.getElementById("application-canvas");
+
+        // Create the application and start the update loop
+        var app = new pc.Application(canvas);
+        app.start();
+
+        // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+        app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+        app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+        window.addEventListener("resize", function () {
+            app.resizeCanvas(canvas.width, canvas.height);
+        });
+
+        var miniStats = new pcx.MiniStats(app);
+
+        app.scene.ambientLight = new pc.Color(0.2, 0.2, 0.2);
+
+        var entity, light, camera;
+
+        // Create an Entity with a camera component
+        var camera = new pc.Entity();
+        camera.addComponent("camera", {
+            clearColor: new pc.Color(0.4, 0.45, 0.5)
+        });
+        camera.translate(0, 0, 15);
+
+        // Create an Entity with a point light component and a sphere model component.
+        var light = new pc.Entity();
+        light.addComponent("light", {
+            type: "point",
+            color: new pc.Color(1, 1, 1),
+            range: 30
+        });
+        light.translate(5, 5, 10);
+
+        app.root.addChild(camera);
+        app.root.addChild(light);
+
+        // Create a texture to hold the video frame data
+        var videoTexture = new pc.Texture(this.app.graphicsDevice, {
+            format: pc.PIXELFORMAT_R5_G6_B5,
+            autoMipmap: false
+        });
+        videoTexture.minFilter = pc.FILTER_LINEAR;
+        videoTexture.magFilter = pc.FILTER_LINEAR;
+        videoTexture.addressU = pc.ADDRESS_CLAMP_TO_EDGE;
+        videoTexture.addressV = pc.ADDRESS_CLAMP_TO_EDGE;
+
+        // Grab our HTML element with the video
+        var video = document.getElementById('vid');
+        video.addEventListener('canplay', function (e) {
+            videoTexture.setSource(video);
+        });
```

Comment:
This will need to be changed slightly to work on iOS. See PR here: https://github.com/playcanvas/developer.playcanvas.com/pull/240

Reply:
I added it to the asset load callback. If I understood correctly from MDN, it resets the state of the media element and prepares it for playback.

Reply:
Ah right, I see it now. You shouldn't need to load the video asset as the video tag will stream it directly to the player

Reply:
Right, but I am loading a TV model asset, which is a GLB file containing a mesh that will display the video texture.

Reply:
Gotcha, sounds good!